src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUnary.cs

```diff
@@ -32,14 +32,14 @@ internal static partial class Library
             },
             {
                 UnaryOpKind.NegateDecimal,
-                StandardErrorHandling<DecimalValue>(
+                StandardErrorHandling<FormulaValue>(
                     "-",
                     expandArguments: NoArgExpansion,
                     replaceBlankValues: ReplaceBlankWithDecimalZero,
-                    checkRuntimeTypes: ExactValueType<DecimalValue>,
+                    checkRuntimeTypes: DateNumberTimeOrDateTime,
```

Comment:
Any reason to do this at runtime rather than pick between them in the IRTranslator? We know statically that we're negating a number vs a date, vs a time. 
Adding UnaryOpKinds for each of those would likely be cleaner, and would allow backends like SQL to handle them more easily and check that they're handling everything they need to.

Reply:
Updated it to use new unary op kinds